src/fcmp_pp/fcmp_pp_crypto.cpp

```diff
@@ -304,6 +304,35 @@ rct::key clear_torsion(const ge_p3 &point) {
     return k_out;
 }
 //----------------------------------------------------------------------------------------------------------------------
+bool get_valid_torsion_cleared_point(const rct::key &point, rct::key &torsion_cleared_out) {
+    ge_p3 p3;
+    if (ge_frombytes_vartime(&p3, point.bytes) != 0)
+        return false;
+    if (mul8_is_identity(p3))
+        return false;
+    torsion_cleared_out = fcmp_pp::clear_torsion(p3);
+    if (torsion_cleared_out == rct::I)
+        return false;
+    return true;
```

Comment:
```suggestion
    if (!rct::isInMainSubgroup(point))
        return false;
    torsion_cleared_out = point;
    return true;
```

`rct::isInMainSubgroup` / `rct::toPointCheckOrder` under the hood works a bit differently than `clear_torsion`. They multiply by the order of the prime subgroup `rct::curveOrder()` and do a positive check against the additive identity element `rct::I`. Is this essentially the same operation? Especially thinking about points with order `2l`, `4l`, and `8l`.

Reply:
A core difference is that given a torsioned output, `clear_torsion` will actually clear torsion on that torsioned point and return that "torsion cleared" output, which is intended since technically even outputs with torsion are supposed remain spendable after FCMP++ (but outputs added to the tree cannot have torsion). I believe all outputs with torsion would fail `rct::isInMainSubgroup`, and then wouldn't be added to the tree with this code

Reply:
E.g. check this modification to the `genesis_tx_output_torsion` unit test:

```cpp
TEST(Crypto, genesis_tx_output_torsion)
{
  rct::key k;
  // see config::GENESIS_TX
  epee::string_tools::hex_to_pod("9b2e4c0281c0b02e7c53291a94d1d0cbff8883f8024f5142ee494ffbbd088071", k);
  ge_p3 x;
  ASSERT_EQ(ge_frombytes_vartime(&x, k.bytes), 0);
  EXPECT_FALSE(rct::isInMainSubgroup(k));
  const rct::key cleared = fcmp_pp::clear_torsion(x);
  ASSERT_NE(k, cleared);
}
```